back-end/java/main/java/src/DormRoom/DormRoom.java

```diff
@@ -0,0 +1,157 @@
+package src.DormRoom;
+
+/**
+ * The {@code DormRoom} class represents a dormitory room with various attributes such as size,
+ * capacity, building location, and amenities. It is designed to encapsulate data about a single
+ * dorm room, including its building, the type of bathroom it has, and whether it includes a kitchen
+ * or is part of a suite. Instances of this class are immutable once constructed.
+ *
+ * <p>Usage example:
+ * <pre>
+ * DormRoom myRoom = new DormRoom(200, 255, 2,
+ *    'example.com', true, false,
+ *    BathroomType.Private, "Barbour Hall");
+ * </pre>
+ */
+public class DormRoom {
+
+    /**
+     * The square footage of the room.
+     */
+    final int roomSize;
+
+    /**
+     * The room number, as assigned by the Office of Residential Life. For example, "Room 255".
+     */
+    final int roomNumber;
+
+    /**
+     * The number of people that can reside in this dorm room.
+     */
+    final int roomCapacity;
+
+    /**
+     * A link (URL) to the floor plan that includes this room.
+     */
+    final String floorPlanLink;
+
+    /**
+     * Indicates whether the room has a kitchen.
+     */
+    final boolean hasKitchen;
+
+    /**
+     * Indicates whether the room is considered a suite.
+     */
+    final boolean isSuite;
+
+    /**
+     * The type of bathroom available for this room.
+     */
+    final BathroomType bathroomType;
+
+    /**
+     * The building in which this dorm room is located.
+     */
+    final DormBuilding dormBuilding;
+
+    /**
+     * Constructs a new {@code DormRoom} with the given attributes.
```

Comment:
One possible additional user story: Since we have access to the live csv, it should tell us which rooms are sub-free / single gender / etc. We could later add sorting for this data as well (although this will probably be complicated on the frontend, as its adding a mostly unnecessary variable)